### PR TITLE
feat: integrate AWS cur to primary cluster

### PR DESCRIPTION
**Pull request by falgofrancis**

### CODE REVIEW

File: terraform/aws-multi-cluster/aws_iam.tf [hunk 1]
```diff
@@ -10,6 +10,12 @@ resource "aws_iam_role" "kubecost_federated_storage" {
   })
 
   tags = var.tags
+
+  lifecycle {
+    ignore_changes = [
+      role_last_used,
+    ]
+  }
```
Comment (thomasvn) [inline terraform/aws-multi-cluster/aws_iam.tf:18]: Thanks for adding this. Could you clarify why it's needed? If multiple users/roles touch the bucket, does it update the bucket's `role_last_used` attribute?

File: terraform/aws-multi-cluster/aws_s3.tf [hunk 2]
```diff
@@ -1,60 +1,87 @@
...
-            "Resource": [
-                "arn:aws:s3:::${var.federated_storage_bucket}/*",
-                "arn:aws:s3:::${var.federated_storage_bucket}"
-            ]
-        }
-    ]
+  rule {
+    id = "expire"
+    expiration {
+      days = var.federated_storage_days
```
Comment (thomasvn) [inline terraform/aws-multi-cluster/aws_s3.tf:26]: 👍🏼, nice!

File: terraform/aws-multi-cluster/crawler-cfn.yml [hunk 3]
```diff
@@ -50,7 +50,7 @@ Resources:
                 Action:
                   - 's3:GetObject'
                   - 's3:PutObject'
-                Resource: !Sub 'arn:$${AWS::Partition}:s3:::${CUR_BUCKET}/${CUR_PREFIX}/${ACCOUNT_NUM}*'
+                Resource: !Sub 'arn:$${AWS::Partition}:s3:::${CUR_BUCKET}/${CUR_PREFIX}/*'
```
Comment (thomasvn) [inline terraform/aws-multi-cluster/crawler-cfn.yml:53]: @falgofrancis Have you confirmed this works in your environment?

**Review by thomasvn — APPROVED** [review]
Thanks for polishing this, looks good to me!

File: terraform/aws-multi-cluster/aws_iam.tf (hunk 1 above)
Comment (falgofrancis) [reply terraform/aws-multi-cluster/aws_iam.tf:18]: Whenever I apply terraform, terraform was showing it as diff and re applying. I did not want to reapply for this particular diff

File: terraform/aws-multi-cluster/crawler-cfn.yml (hunk 3 above)
Comment (falgofrancis) [reply terraform/aws-multi-cluster/crawler-cfn.yml:53]: yes, we have CUR data from single folder.

**Pull request merged**